internal/versions/versions.go

```diff
@@ -26,17 +26,22 @@ type Versions struct {
 //go:generate mockgen -package versions -destination mock_versions.go -self_package github.com/openshift/assisted-service/internal/versions . Handler
 type Handler interface {
 	restapi.VersionsAPI
-	GetReleaseImage(openshiftVersion string) (string, error)
-	GetRHCOSImage(openshiftVersion string) (string, error)
-	GetRHCOSRootFS(openshiftVersion string) (string, error)
-	GetRHCOSVersion(openshiftVersion string) (string, error)
-	GetReleaseVersion(openshiftVersion string) (string, error)
+	GetReleaseImage(openshiftVersion, cpuArchitecture string) (string, error)
+	GetReleaseVersion(openshiftVersion, cpuArchitecture string) (string, error)
+	GetRHCOSImage(openshiftVersion, cpuArchitecture string) (string, error)
+	GetRHCOSRootFS(openshiftVersion, cpuArchitecture string) (string, error)
+	GetRHCOSVersion(openshiftVersion, cpuArchitecture string) (string, error)
+	GetCPUArchitectures(openshiftVersion string) ([]string, error)
 	GetKey(openshiftVersion string) (string, error)
-	GetVersion(openshiftVersion string) (*models.OpenshiftVersion, error)
-	IsOpenshiftVersionSupported(versionKey string) bool
+	GetRelease(openshiftVersion, cpuArchitecture string) (string, string, error)
+	IsOpenshiftVersionSupported(versionKey, cpuArchitecture string) bool
 	AddOpenshiftVersion(ocpReleaseImage, pullSecret string) (*models.OpenshiftVersion, error)
 }
 
+const (
+	DefaultCPUArchitecture = "x86_64"
+)
```

Comment:
I think this property should be part of the cluster and `versions` should work only with what it gets from the calling function, it shouldn't be familiar with a default configuration.

Reply:
It's still needed for functions as GetCPUArchitectures (for returning the default arch for older versions), and for IsOpenshiftVersionSupported (for having a default to fallback). We can move the constant though if it makes more sense, maybe in inventory?